lldb/include/lldb/Utility/FileSpecList.h

```diff
@@ -38,7 +65,10 @@ class FileSpecList {
   FileSpecList(FileSpecList &&rhs) = default;
 
   /// Initialize this object from a vector of FileSpecs
-  FileSpecList(std::vector<FileSpec> &&rhs) : m_files(std::move(rhs)) {}
+  FileSpecList(std::vector<FileSpec> &&rhs) {
+    for (auto &fs : rhs)
+      m_files.emplace_back(fs);
```

Comment:
This sidesteps the move semantics and means we're copying again. Why?

Reply:
Do you have a suggestion for how to fix this? Would adding a move constructor to FileSpec address your concern?

Note that FileSpecList is a value type that gets copied a lot, which informs many of the odd design choices in this patch.

Reply:
> Would adding a move constructor to FileSpec address your concern?

No, there's nothing worth move-constructing in FileSpec.

Reply:
FileSpec may not be worth move-constructing, but move-constructing a vector would be worth it right? Maybe I'm missing something here though.

Reply:
Yes: one is a vector<FileSpec> the other a vector<FileSpecHolder>